src/d3a/d3a_core/export.py

```diff
@@ -436,6 +438,55 @@ def _plot_ess_soc_history(self, storage_list: list, subdir: str, root_name: str)
         output_file = os.path.join(plot_dir, 'ess_soc_history_{}.html'.format(root_name))
         PlotlyGraph.plot_bar_graph(barmode, title, xtitle, ytitle, data, output_file)
 
+    def plot_ess_energy_trace(self, area, subdir):
+        """
+        Wrapper for _plot_ess_energy_trace.
+        """
+
+        new_subdir = os.path.join(subdir, area.slug)
+        from d3a.models.strategy.storage import StorageStrategy
+        storage_list = [child for child in area.children
+                        if isinstance(child.strategy, StorageStrategy)]
+        for element in storage_list:
+            self._plot_ess_energy_trace(element.strategy.state.time_series_ess_share,
+                                        new_subdir, area.slug)
+        for child in area.children:
+            if child.children:
+                self.plot_ess_energy_trace(child, new_subdir)
+
+    def _plot_ess_energy_trace(self, energy: dict, subdir: str, root_name: str):
+        """
+        Plots ess energy trace for each knot in the hierarchy
+        """
+
+        storage_key = 'charge [%]'
+        data = list()
+        barmode = "stack"
+        title = 'ESS ENERGY SHARE ({})'.format(root_name)
+        xtitle = 'Time'
+        ytitle = 'Energy [kWh]'
+
+        graph_obj = PlotlyGraph(energy, storage_key)
+        temp = {ESSEnergyOrigin.UNKNOWN: {slot: 0. for slot in generate_market_slot_list()},
+                ESSEnergyOrigin.LOCAL: {slot: 0. for slot in generate_market_slot_list()},
+                ESSEnergyOrigin.EXTERNAL: {slot: 0. for slot in generate_market_slot_list()}}
+
+        for k, v in graph_obj.dataset.items():
```

Comment:
Can you please explain why this step is necessary ? Also, why are you initialising the PlotlyGraph and also go.Bar object? I think that PlotlyGraph.plot_bar_graph() does everything you want. You just have to put the data in the right format.

Reply:
Data is in different format due to which PlotlyGraph.plot_bar_graph() wouldn't be able to work.